File: CS/Q253493/FilterControl/CustomClauseNode.cs
﻿using DevExpress.Data.Filtering;
using DevExpress.XtraEditors.Filtering;

namespace Q253493.FilterControl {
    public class CustomClauseNode :ClauseNode {
        public CustomClauseNode(FilterTreeNodeModel model): base(model) { }

        protected override void ClauseNodeFirstOperandChanged(OperandProperty newProp, int elementIndex) {
            base.ClauseNodeFirstOperandChanged(newProp, elementIndex);
            Operation = ((CustomFilterControl)((WinFilterTreeNodeModel)Model).Control).GetDefaultOperationCore(
                FilterProperties.GetProperty(newProp), Operation);
        }
    }
}

File: VB/Q253493/FilterControl/CustomClauseNode.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.Data.Filtering
Imports DevExpress.XtraEditors.Filtering

Namespace Q253493.FilterControl
	Public Class CustomClauseNode
		Inherits ClauseNode
		Public Sub New(ByVal model As FilterTreeNodeModel)
			MyBase.New(model)
		End Sub

		Protected Overrides Sub ClauseNodeFirstOperandChanged(ByVal newProp As OperandProperty, ByVal elementIndex As Integer)
			MyBase.ClauseNodeFirstOperandChanged(newProp, elementIndex)
			Operation = (CType((CType(Model, WinFilterTreeNodeModel)).Control, CustomFilterControl)).GetDefaultOperationCore(FilterProperties.GetProperty(newProp), Operation)
		End Sub
	End Class
End Namespace
